cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -1152,6 +1152,53 @@ TEST(TestBinaryArithmetic, AddWithImplicitCastsUint64EdgeCase) {
                                      ArrayFromJSON(uint64(), "[18446744073709551615]")}));
 }
 
+TEST(TestBinaryArithmetic, MultiplyDuration) {
```

Comment:
We already have `TestTemporalMultiplyDuration` in `scalar_temporal_test.cc`, can you augment that test instead?

Also, notice `TestTemporalDivideDuration` which hints that the same improvement could be done for division...

Reply:
Oh, I didn't notice this. I've changed the tests.